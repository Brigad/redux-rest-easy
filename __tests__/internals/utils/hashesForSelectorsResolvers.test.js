import {
  computeHashesForSelectorsResolvers,
  getEmptyResourceHash,
  getPayloadIdsHash,
  getResourceHash,
  getResourcesHash,
  resetAllResourcesHashForSelectorsResolvers,
  resetResourceHashForSelectorsResolvers,
} from '../../../src/internals/utils/hashesForSelectorsResolvers';

const RESOURCE_NAME = 'fruits';
const NORMALIZED_URL = 'eat:https://api.co/fruits';
const PRINCIPAL_RESOURCE_IDS = ['2', '3', '1'];
const NORMALIZED_PAYLOAD = {
  [RESOURCE_NAME]: {
    1: {
      type: 'apple',
      juicy: true,
    },
    2: {
      type: 'banana',
      juicy: false,
    },
    3: {
      type: 'cherry',
      juicy: false,
    },
  },
};
const EMPTY_STATE = {};
const STATE = {
  requests: {
    [NORMALIZED_URL]: {
      payloadIds: {
        [RESOURCE_NAME]: PRINCIPAL_RESOURCE_IDS,
      },
    },
  },
  resources: NORMALIZED_PAYLOAD,
};

const computeHashesEmptyState = () => {
  computeHashesForSelectorsResolvers(
    EMPTY_STATE,
    RESOURCE_NAME,
    NORMALIZED_URL,
    NORMALIZED_PAYLOAD,
    PRINCIPAL_RESOURCE_IDS,
  );
};

const computeHashes = () => {
  computeHashesForSelectorsResolvers(
    STATE,
    RESOURCE_NAME,
    NORMALIZED_URL,
    NORMALIZED_PAYLOAD,
    PRINCIPAL_RESOURCE_IDS,
  );
};

describe('computeHashesForSelectorsResolvers', () => {
  afterEach(resetAllResourcesHashForSelectorsResolvers);

  test('empty state', () => {
    const hashBeforeComputing = getResourcesHash();

    computeHashesEmptyState();

    const hashAfterComputing = getResourcesHash();

    expect(hashBeforeComputing).not.toBe(hashAfterComputing);
  });

  test('filled state', () => {
    const hashBeforeComputing = getResourcesHash();

    computeHashes();

    const hashAfterComputing = getResourcesHash();

    expect(hashBeforeComputing).not.toBe(hashAfterComputing);
  });
});

describe('resetResourceHashForSelectorsResolvers', () => {
  afterEach(resetAllResourcesHashForSelectorsResolvers);

  test('only path', () => {
    const hashBeforeComputing = getResourceHash(RESOURCE_NAME);

    computeHashes();

    const hashAfterComputing = getResourceHash(RESOURCE_NAME);

    resetResourceHashForSelectorsResolvers(STATE, RESOURCE_NAME);

    const hashAfterResetting = getResourceHash(RESOURCE_NAME);

    expect(hashBeforeComputing).not.toBe(hashAfterComputing);
    expect(hashBeforeComputing).toBe(hashAfterResetting);
  });
});

describe('resetAllResourcesHashForSelectorsResolvers', () => {
  afterEach(resetAllResourcesHashForSelectorsResolvers);

  test('only path', () => {
    const hashBeforeComputing = getResourceHash(RESOURCE_NAME);

    computeHashes();

    const hashAfterComputing = getResourceHash(RESOURCE_NAME);

    resetAllResourcesHashForSelectorsResolvers();

    const hashAfterResetting = getResourceHash(RESOURCE_NAME);

    expect(hashBeforeComputing).not.toBe(hashAfterComputing);
    expect(hashBeforeComputing).toBe(hashAfterResetting);
  });
});

describe('getEmptyResourceHash', () => {
  afterEach(resetAllResourcesHashForSelectorsResolvers);

  test('without computing first', () => {
    expect(getEmptyResourceHash()).toMatchSnapshot();
  });

  test('after computing', () => {
    computeHashes();

    expect(getEmptyResourceHash()).toMatchSnapshot();
  });
});

describe('getPayloadIdsHash', () => {
  afterEach(resetAllResourcesHashForSelectorsResolvers);

  test('no normalizedURL', () => {
    expect(getPayloadIdsHash()).toMatchSnapshot();
  });

  test('no resourceName', () => {
    expect(getPayloadIdsHash(NORMALIZED_URL)).toMatchSnapshot();
  });

  test('without computing first', () => {
    expect(getPayloadIdsHash(NORMALIZED_URL, RESOURCE_NAME)).toMatchSnapshot();
  });

  test('after computing', () => {
    computeHashes();

    expect(getPayloadIdsHash(NORMALIZED_URL, RESOURCE_NAME)).toMatchSnapshot();
  });
});

describe('getResourcesHash', () => {
  afterEach(resetAllResourcesHashForSelectorsResolvers);

  test('without computing first', () => {
    expect(getResourcesHash()).toMatchSnapshot();
  });

  test('after computing', () => {
    computeHashes();

    expect(getResourcesHash()).toMatchSnapshot();
  });
});

describe('getResourceHash', () => {
  afterEach(resetAllResourcesHashForSelectorsResolvers);

  test('no resourceName', () => {
    expect(getResourceHash()).toMatchSnapshot();
  });

  test('without computing first', () => {
    expect(getResourceHash(RESOURCE_NAME)).toMatchSnapshot();
  });

  test('after computing', () => {
    computeHashes();

    expect(getResourceHash(RESOURCE_NAME)).toMatchSnapshot();
  });
});
