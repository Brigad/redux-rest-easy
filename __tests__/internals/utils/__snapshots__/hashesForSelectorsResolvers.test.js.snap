// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEmptyResourceHash after computing 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;

exports[`getEmptyResourceHash without computing first 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;

exports[`getPayloadIdsHash after computing 1`] = `"f00555c74464ef36bbc004bcf654aca1"`;

exports[`getPayloadIdsHash no normalizedURL 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;

exports[`getPayloadIdsHash no resourceName 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;

exports[`getPayloadIdsHash without computing first 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;

exports[`getResourceHash after computing 1`] = `"2d64c2982ff64c3288c39158728ba72c"`;

exports[`getResourceHash no resourceName 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;

exports[`getResourceHash without computing first 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;

exports[`getResourcesHash after computing 1`] = `"529477faa63b29b600bab57fb91ba873"`;

exports[`getResourcesHash without computing first 1`] = `"cf252fdcd0c577918a558e3d5869780d"`;
